### PR TITLE
Add note about specifying custom base class for Exceptions.

### PR DESCRIPTION
**Pull request by michalsustr**

### CODE REVIEW

File: docs/advanced/exceptions.rst [hunk 1]
```diff
@@ -79,6 +79,20 @@ This call creates a Python exception class with the name ``PyExp`` in the given
 module and automatically converts any encountered exceptions of type ``CppExp``
 into Python exceptions of type ``PyExp``.
 
+It is possible to specify base class for the exception using the third 
+parameter, a pointer to `PyObject`:
```
Comment (YannickJadoul) [inline docs/advanced/exceptions.rst:83]: I know this is currently correct, but do we have any idea why this isn't a `handle`?

File: docs/advanced/exceptions.rst [hunk 2]
```diff
@@ -79,6 +79,20 @@ This call creates a Python exception class with the name ``PyExp`` in the given
...
+.. code-block:: cpp
+
+    py::register_exception<CppExp>(module, "PyExp", PyExc_RuntimeError);
+
+Then `PyExp` can be caught both as `PyExp` and `RuntimeError`.
+Common base classes from the table above are `PyExc_Exception` (default), 
+`PyExc_IndexError`, `PyExc_KeyError`, `PyExc_MemoryError`, `PyExc_OverflowError`, 
+`PyExc_RuntimeError`, `PyExc_StopIteration`, `PyExc_ValueError`. 
+More definitions can be found in Python's `pyerrors.h` file, and they follow
+the pattern `PyExc_XXX`.
```
Comment (YannickJadoul) [inline docs/advanced/exceptions.rst:94]: Maybe we can just refer to the Python docs? Those are more accessible than `pyerror.h`, and then this part can be smaller (i.e. "The class objects of the built-in Python exceptions are listed in the Python documentation:" or so) : https://docs.python.org/3/c-api/exceptions.html#standard-exceptions

File: docs/advanced/exceptions.rst [hunk 3]
```diff
@@ -79,6 +79,20 @@ This call creates a Python exception class with the name ``PyExp`` in the given
...
 into Python exceptions of type ``PyExp``.
 
+It is possible to specify base class for the exception using the third 
+parameter, a pointer to `PyObject`:
+
+.. code-block:: cpp
+
+    py::register_exception<CppExp>(module, "PyExp", PyExc_RuntimeError);
+
+Then `PyExp` can be caught both as `PyExp` and `RuntimeError`.
```
Comment (YannickJadoul) [inline docs/advanced/exceptions.rst:89]: Does this need to be explicitly specified? This is really just the basic way Python handles an exception hierarchy. (Fine to keep it though, if you want.)

**Review by YannickJadoul — COMMENTED** [review]
Thanks, @michalsustr! A few minor remarks, but good to have this documented :-)

File: docs/advanced/exceptions.rst (hunk 3 above)
Comment (michalsustr) [reply docs/advanced/exceptions.rst:89]: I'd like to keep it, even though it's basic (makes sure there is common understanding)

File: docs/advanced/exceptions.rst (hunk 1 above)
Comment (michalsustr) [reply docs/advanced/exceptions.rst:83]: No idea, first time contributing to pybind.

Comment (YannickJadoul) [reply docs/advanced/exceptions.rst:83]: No problem. Let's see if @bstaletic or anyone else knows! :-)

File: docs/advanced/exceptions.rst (hunk 3 above)
Comment (YannickJadoul) [reply docs/advanced/exceptions.rst:89]: OK, fine with me.

File: docs/advanced/exceptions.rst [hunk 4]
```diff
@@ -79,6 +79,19 @@ This call creates a Python exception class with the name ``PyExp`` in the given
 module and automatically converts any encountered exceptions of type ``CppExp``
 into Python exceptions of type ``PyExp``.
 
+It is possible to specify base class for the exception using the third
```
Comment (YannickJadoul) [inline docs/advanced/exceptions.rst:82]: The trailing space on this line is causing our CI to fail it's "Format" job.

**Review by YannickJadoul — APPROVED** [review]
Looks good to me, once the CI gets fixed! :-)

File: docs/advanced/exceptions.rst (hunk 4 above)
Comment (michalsustr) [reply docs/advanced/exceptions.rst]: It is indeed formatting very well :D fixed

Comment (YannickJadoul) [reply docs/advanced/exceptions.rst]: A little nitpicky, but now it's passing :-)

**Review by henryiii — APPROVED** [review]

File: docs/advanced/exceptions.rst (hunk 1 above)
Comment (bstaletic) [reply docs/advanced/exceptions.rst:83]: Nope, no clue other than "the source says so".

**Review by bstaletic — APPROVED** [review]
Looks good.

File: docs/advanced/exceptions.rst (hunk 1 above)
Comment (YannickJadoul) [reply docs/advanced/exceptions.rst:83]: OK, let's maybe change this in a different PR, then? (Actually, we might even change this to `py::type`, @henryiii?)

**Comment by YannickJadoul** [comment]
No further comments, it seems, so let's merge. (If anyone still has remarks, afterwards, it's just docs and easy enough to change.)

**Comment by YannickJadoul** [comment]
Thanks again, @michalsustr!

**Pull request merged**